src/test/blockencodings_tests.cpp

```diff
@@ -207,7 +207,7 @@ BOOST_AUTO_TEST_CASE(NonCoinbasePreforwardRTTest)
 
 BOOST_AUTO_TEST_CASE(SufficientPreforwardRTTest)
 {
-    CTxMemPool pool;
+    CTxMemPool& pool = *Assert(m_node.mempool);
```

Comment:
I'm somewhat divided about this change. From a "testing hygiene" point of view it seems preferable to me to construct a new mempool instead of use an existing one from the global state.

I also like the idea of a function to construct a default mempool for testing. But maybe I'm missing something.

Reply:
> From a "testing hygiene" point of view it seems preferable to me to construct a new mempool instead of use an existing one from the global state.

In cases where we only need one mempool that does not outlive `TestingSetup` (the cases addressed in this PR), it is not a problem to use `TestingSetup`'s mempool since it's reconstructed each time. In many cases it is much more preferable to use `TestingSetup`'s mempool:

- `TestingSetup`'s mempool will respect `gArgs` flags given to it by its constructor, local mempools will not. This is an easier footgun to shoot than one would think, and has already led to a case where we were fuzzing a noop: #24634
- `TestingSetup`'s mempool is registered in `TestingSetup`'s `CChainState`s' `m_mempool` and perhaps in other places. Using a local mempool means that there will be a discrepancy between the mempool the test operates on and the mempool that `CChainState`s operate on.

To be clear, I'm not saying that all of our tests all have to use `TestingSetup`'s mempool, there are many places where that doesn't make sense (see this PR's original description). I'm only changing places where it's obviously correct to use `TestingSetup`'s mempool.

> I also like the idea of a function to construct a default mempool for testing.

Sure! I like it too. It's great for cases of `CTxMemPool` construction that this PR deliberately leaves out. It can even use the `DummyChainstate` technique in `src/test/fuzz/tx_pool.cpp` to replace the `m_mempool` pointer in `CChainState`. It's just done easier after we introduce `CTxMemPool::Options` in #25290 so I think we can table it for now.

Reply:
OK, agree. I was afraid this would introduce errors the other way around (by accidentally reusing mempool between tests), but if that's not the case, and can even avoid unexpected behavior, great.